packages/preview/src/browser/preview-contribution.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { FrontendApplicationContribution, FrontendApplication, OpenHandler, ApplicationShell } from "@theia/core/lib/browser";
+import { EDITOR_CONTEXT_MENU, EditorManager, TextEditor } from '@theia/editor/lib/browser';
+import { CommandContribution, CommandRegistry, Command, MenuContribution, MenuModelRegistry, CommandHandler, Disposable } from "@theia/core/lib/common";
+import { DisposableCollection } from '@theia/core';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import URI from '@theia/core/lib/common/uri';
+import { Position } from 'vscode-languageserver-types';
+import { PreviewWidget, PREVIEW_WIDGET_FACTORY_ID } from './preview-widget';
+import { PreviewHandlerProvider } from './preview-handler';
+
+export namespace PreviewCommands {
+    export const OPEN: Command = {
+        id: 'preview:open',
+        label: 'Open Preview'
+    };
+}
+
+@injectable()
+export class PreviewContribution implements CommandContribution, MenuContribution, OpenHandler, FrontendApplicationContribution {
+
+    readonly id = 'preview';
+    readonly label = 'Preview';
+
+    protected readonly disposables = new DisposableCollection();
+    protected previewWidget: PreviewWidget | undefined;
```

Comment:
it should not be necessary, you can always get it from the widget manager, right?

Reply:
sounds good.